plugins/catalog/src/components/SystemDiagram/SystemDiagram.tsx

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Entity } from '@backstage/catalog-model';
+import { catalogApiRef } from '@backstage/plugin-catalog-react';
+import {
+  DependencyGraph,
+  DependencyGraphTypes,
+  InfoCard,
+  Progress,
+  useApi,
+} from '@backstage/core';
+import { Alert } from '@material-ui/lab';
+import React from 'react';
+import { useAsync } from 'react-use';
+
+type SystemDiagramProps = {
+  entity: Entity;
+  variant?: string;
+};
+
+export function SystemDiagram({ entity }: SystemDiagramProps) {
+  const catalogApi = useApi(catalogApiRef);
+  const { loading, error, value: catalogResponse } = useAsync(() => {
+    return catalogApi.getEntities({
+      filter: {
+        kind: ['Component', 'API', 'Resource', 'System', 'Domain'],
+      },
+    });
+  }, [catalogApi]);
+
+  const currentSystemName = entity.metadata.name;
+  const currentSystemNode = `system:${entity.metadata.name}`.toLowerCase();
+  const systemNodes = [];
+  const systemEdges = [];
+
+  if (catalogResponse && catalogResponse.items) {
+    for (const catalogItem of catalogResponse.items) {
+      // pick out the system itself
+      if (catalogItem.metadata.name === entity.metadata.name) {
+        systemNodes.push({
+          id: currentSystemNode,
+        });
+
+        // check if the system has an assigned domain
+        // even if the domain object doesn't exist in the catalog, display it in the map
+        if (catalogItem.spec?.domain) {
```

Comment:
Remember to not read out of the spec - use `getEntityRelations` out of `@backstage/plugin-catalog-react` to get the proper relation. Something like `getEntityRelations(catalogItem, RELATION_PART_OF, { kind: 'domain' }`

After that I recommend using `compareEntityToRef` out of `@backstage/catalog-model` to make sure that the comparison happens case insensitively etc